### PR TITLE
Update BDP Cloud docs

### PR DESCRIPTION
**Pull request by lkrystkowiak-snowplow**

### CODE REVIEW

**Comment by lkrystkowiak-snowplow** [comment]
cc @ebedford420

**Comment by lkrystkowiak-snowplow** [comment]
cc @elliotstokes

**Review by stanch — CHANGES_REQUESTED** [review]
Run “git grep Preview” — there are a few more references to Cloud Preview

File: docs/recipes/recipe-content-analytics/index.md [hunk 1]
```diff
@@ -27,7 +27,7 @@ We have already created a custom `content` entity for you in [Iglu Central](http
 
 Snowplow uses self-describing JSON schemas to structure events and entities so that they can be validated in the pipeline and loaded into tidy tables in the warehouse. You can learn more about these data structures [here](/docs/understanding-tracking-design/understanding-schemas-and-validation/index.md), and about why we take this approach [here](https://snowplowanalytics.com/blog/2020/01/24/re-thinking-the-structure-of-event-data/).
 
-While Try Snowplow and BDP Cloud Preview only ship with a pre-designed set of custom events and entities required for the recipes, Snowplow BDP lets you create an unlimited number of your own via the [Data Structures UI](/docs/understanding-tracking-design/managing-data-structures/index.md) (and API). Custom events and entities will come soon to BDP Cloud.
+While Try Snowplow only ships with a pre-designed set of custom events and entities required for the recipes, Snowplow BDP lets you create an unlimited number of your own via the [Data Structures UI](/docs/understanding-tracking-design/managing-data-structures/index.md) (and API).
```
Comment (stanch) [inline docs/recipes/recipe-content-analytics/index.md:30]: Should this be pointing to (not yet existing) docs on the schema builder for BDP Cloud?

Comment (lkrystkowiak-snowplow) [reply docs/recipes/recipe-content-analytics/index.md:30]: yup, I think so. Gotta discuss that with @elyerry

File: docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+---
+title: "🆕 Managing data structures with the Data Structures Builder"
+date: "2023-03-01"
```
Comment (stanch) [inline docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md:3]: Not needed for new pages

File: docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+![](images/data-structures-1.png)
+
+Populate the general information, such as *Name,* and a *Description* and *Vendor*. *Vendor* allows you to organise your data structures, for example, by teams. Snowplow will automatically generate the Tracking Url to be referenced in your tracking code.
+
+You can then add one or multiple properties. For each property, you can set a name, description, its type, a possible enumeration of allowed values. You can also set additional constraints to define if this property should be optional or mandatory, and if *null* values are allowed.
+
+![](images/data-structures-2.png)
+
+Click save on the Property modal and this will save your property changes.
```
Comment (stanch) [inline docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md:21]:
```suggestion
Click “Save” on the Property dialog box and this will save your property changes.
```
Or any other replacement of “modal” (which is a very cryptic term for user-facing documentation)

File: docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+Clicking Save on the Data Structure page will publish your Data Structure to your Iglu Server.
```
Comment (stanch) [inline docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md:23]:
```suggestion
Clicking “Save” on the Data Structure page will publish your Data Structure to your Iglu Server.
```

File: docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+:::info
+The Data Structures Builder is only available for BDP Cloud Customers
+:::
+
+To create a new Data Structure, first navigate to **Data Structures** in the menu and click the **'Create a Data Structure'** button.
+
+![](images/data-structures-1.png)
+
+Populate the general information, such as *Name,* and a *Description* and *Vendor*. *Vendor* allows you to organise your data structures, for example, by teams. Snowplow will automatically generate the Tracking Url to be referenced in your tracking code.
```
Comment (stanch) [inline docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md:15]: This does not seem to offer much guidance beyond what the UI already expects the user to do. Can we give some examples here, using a realistic data structure someone might want to add?

File: docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+title: "🆕 Managing data structures with the Data Structures Builder"
+date: "2023-03-01"
+sidebar_position: 85
+---
+
+:::info
+The Data Structures Builder is only available for BDP Cloud Customers
+:::
+
+To create a new Data Structure, first navigate to **Data Structures** in the menu and click the **'Create a Data Structure'** button.
```
Comment (stanch) [inline docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md:11]: This feels like it’s missing some link to other sections explaining what data structures are for (just imagining someone landing here with no context).

File: docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+![](images/data-structures-1.png)
+
+Populate the general information, such as *Name,* and a *Description* and *Vendor*. *Vendor* allows you to organise your data structures, for example, by teams. Snowplow will automatically generate the Tracking Url to be referenced in your tracking code.
+
+You can then add one or multiple properties. For each property, you can set a name, description, its type, a possible enumeration of allowed values. You can also set additional constraints to define if this property should be optional or mandatory, and if *null* values are allowed.
```
Comment (stanch) [inline docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md:17]: Examples? Considerations for why something should be mandatory or optional, and when should `null`s be allowed?

File: docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+![](images/data-structures-2.png)
+
+Click save on the Property modal and this will save your property changes.
+
+Clicking Save on the Data Structure page will publish your Data Structure to your Iglu Server.
+
+For information about editing and versioning your Data Structure please see the [Versioning in BDP Cloud section](/docs/understanding-tracking-design//versioning-your-data-structures/cloud/index.md).
```
Comment (stanch) [inline docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md:25]:
```suggestion
For information about editing and versioning your Data Structure, please see [Versioning](/docs/understanding-tracking-design/versioning-your-data-structures/cloud/index.md) in BDP Cloud section.
```

File: docs/understanding-tracking-design/predefined-vs-custom-entities/index.md [hunk 9]
```diff
@@ -23,6 +23,10 @@ window.snowplow("newTracker", "sp", "{{COLLECTOR_URL}}", {
 
 ## Custom contexts
 
+:::info
+Please note that for BDP Cloud customers this complexity is abstracted by [the Data Structures Builder](/docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md).
```
Comment (stanch) [inline docs/understanding-tracking-design/predefined-vs-custom-entities/index.md:27]: What does this mean? What do we expect the users to do with this information? I would rephrase to something more actionable, like:
```suggestion
If you are using BDP Cloud, you can create custom schemas using the [Data Structures Builder](/docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md) without worrying about how it works under the hood.
```

File: docs/understanding-tracking-design/understanding-schemas-and-validation/index.md [hunk 10]
```diff
@@ -27,6 +27,10 @@ Rather than just leaving your data open to interpretation by the many different
 
 ## The anatomy of a schema
 
+:::info
+Please note that for BDP Cloud customers this complexity is abstracted by [the Data Structures Builder](/docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md).
```
Comment (stanch) [inline docs/understanding-tracking-design/understanding-schemas-and-validation/index.md:31]: Same comment as above

File: docs/understanding-tracking-design/versioning-your-data-structures/cloud/index.md [hunk 11]
```diff
@@ -0,0 +1,22 @@
+---
+title: "Versioning Data Structures in BDP Cloud"
+date: "2023-03-01"
+sidebar_position: 10
+---
+
+# Versioning in BDP Cloud
+
+If you are a BDP Cloud customer, you will have used our Data Structure Builder to create your custom Data Structure for your Events and Entities.
```
Comment (stanch) [inline docs/understanding-tracking-design/versioning-your-data-structures/cloud/index.md:9]:
```suggestion
If you are a BDP Cloud customer, you will have used our Data Structure Builder to create your custom Data Structures for your Events and Entities.
```

File: docs/understanding-tracking-design/versioning-your-data-structures/cloud/index.md [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+
+![](../../managing-data-structures-with-data-structures-builder/images/data-structures-2.png)
+
+**Breaking and non-breaking changes**
+
+At the point of publishing a schema, there are two options for versioning your data structure.
```
Comment (stanch) [inline docs/understanding-tracking-design/versioning-your-data-structures/cloud/index.md:15]: I find the alternate use of terms “schema” and “data structure” confusing. Do they mean the same thing, or do we refer to a specific version of a data structure as a “schema”?

File: docs/understanding-tracking-design/versioning-your-data-structures/cloud/index.md [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+- **Non-breaking** - a non-breaking change is backward compatible with historical data and your loader.
```
Comment (stanch) [inline docs/understanding-tracking-design/versioning-your-data-structures/cloud/index.md:16]:
```suggestion
- **Non-breaking** - a non-breaking change is backward compatible with historical data and your warehouse loader.
```

Comment (stanch) [reply docs/understanding-tracking-design/versioning-your-data-structures/cloud/index.md:16]: Or something that clarifies what we are talking about.

File: docs/understanding-tracking-design/versioning-your-data-structures/cloud/index.md [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+- **Breaking** - a breaking change is not backwards compatible with historical data and/or your loader.
+
+The Data Structure builder will automatically select how to version up your data structure depending on the changes you have just made.
```
Comment (stanch) [inline docs/understanding-tracking-design/versioning-your-data-structures/cloud/index.md:19]: Thinking as a user: “this is good to know... but what does this mean?” Can we explain that non-breaking changes get a minor version bump and major changes get a major version bump? And then what the implications are in the warehouse? There are a lot of assumptions here that we are glossing over and that would be completely unfamiliar to new users, even more so to the intended audience of BDP Cloud.

File: docs/understanding-tracking-design/versioning-your-data-structures/enterprise/index.md [hunk 15]
```diff
@@ -0,0 +1,32 @@
+---
+title: "Versioning Data Structures in BDP Enterprise"
```
Comment (stanch) [inline docs/understanding-tracking-design/versioning-your-data-structures/enterprise/index.md:2]: Do we really need two different pages? Can this be a single page with two sections, or with tabs? I imagine that in near future we might bring some of the Cloud functionality into Enterprise, and then two separate pages will no longer make sense.

File: docs/understanding-tracking-design/versioning-your-data-structures/enterprise/index.md [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+sidebar_position: 20
+---
+
+## How do I version?
+
+### Breaking and non-breaking changes
+
+In Data Structures UI at the point of publishing a schema you'll be asked to select which version you'd like to create. There are two options:
+
+- **Non-breaking** - a non-breaking change is backward compatible with historical data and increments the `patch` number i.e. `1-0-0` -> `1-0-1`.
```
Comment (stanch) [inline docs/understanding-tracking-design/versioning-your-data-structures/enterprise/index.md:13]: This explanation seems like it could work for both Cloud and Enterprise. Why duplicate it?

File: docs/understanding-tracking-design/versioning-your-data-structures/index.md [hunk 17]
```diff
@@ -14,29 +14,8 @@ As well as good practice, versioning has an important role in telling Snowplow L
 
 For example, for certain changes there will be a need to create new columns, update columns or even create whole new tables. For this reason, it's important you understand when your change is breaking and version correctly.
```
Comment (stanch) [inline docs/understanding-tracking-design/versioning-your-data-structures/index.md:15]: Once again, I think this should be kept as a single page, perhaps with tabs for Cloud and Enterprise and some common text.

**Review by stanch — CHANGES_REQUESTED** [review]

File: docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md (hunk 5 above)
Comment (elyerry) [reply docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md:15]: We're going to add a video tutorial soon where we go through an actual example.

File: docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md (hunk 6 above)
Comment (elyerry) [reply docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md:11]: this page sits in the Introduction to Tracking Design section, where they can find this into... theoretically they go through each section in order

File: docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md (hunk 7 above)
Comment (elyerry) [reply docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md:17]: These are all good points but they shouldn't really sit in the DS Builder page. They are more Tracking Design principles and guidance.

File: docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md (hunk 8 above)
Comment (elyerry) [reply docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md:25]: 👍

File: docs/understanding-tracking-design/versioning-your-data-structures/enterprise/index.md (hunk 15 above)
Comment (elyerry) [reply docs/understanding-tracking-design/versioning-your-data-structures/enterprise/index.md:2]: We're trying to hide the complexity of versioning Json for Cloud users atm. And besides the current docs are not great, they don't cover all changes. So I'd prefer to leave it separate for the time being and return to it later on.

File: docs/understanding-tracking-design/versioning-your-data-structures/enterprise/index.md (hunk 16 above)
Comment (elyerry) [reply docs/understanding-tracking-design/versioning-your-data-structures/enterprise/index.md:13]: We don't offer patching in Cloud. And you don't get asked what version to pick. It's a different workflow.

File: docs/understanding-tracking-design/versioning-your-data-structures/index.md (hunk 17 above)
Comment (elyerry) [reply docs/understanding-tracking-design/versioning-your-data-structures/index.md:15]: We can agree that these sections need reviewing, but this content as it is does not entirely apply to Cloud customers. And I am wary of just rashly splitting it up. The docs need a better strategy to handle Cloud and Enterprise experiences.

File: docs/understanding-tracking-design/versioning-your-data-structures/cloud/index.md (hunk 14 above)
Comment (elyerry) [reply docs/understanding-tracking-design/versioning-your-data-structures/cloud/index.md:19]: We have a video coming soon where we're showing all that. We are just not quite ready yet to publish it.

File: docs/understanding-tracking-design/versioning-your-data-structures/cloud/index.md (hunk 13 above)
Comment (elyerry) [reply docs/understanding-tracking-design/versioning-your-data-structures/cloud/index.md:16]: 👍

File: docs/understanding-tracking-design/predefined-vs-custom-entities/index.md (hunk 9 above)
Comment (elyerry) [reply docs/understanding-tracking-design/predefined-vs-custom-entities/index.md:27]: 👍

File: docs/understanding-tracking-design/understanding-schemas-and-validation/index.md (hunk 10 above)
Comment (elyerry) [reply docs/understanding-tracking-design/understanding-schemas-and-validation/index.md:31]: 👍

File: docs/understanding-tracking-design/versioning-your-data-structures/cloud/index.md (hunk 11 above)
Comment (elyerry) [reply docs/understanding-tracking-design/versioning-your-data-structures/cloud/index.md:9]: 👍

File: docs/understanding-tracking-design/versioning-your-data-structures/cloud/index.md (hunk 12 above)
Comment (elyerry) [reply docs/understanding-tracking-design/versioning-your-data-structures/cloud/index.md:15]: Think this is an oversight... let's just use Data Structure rather than Schema

File: docs/recipes/recipe-content-analytics/index.md (hunk 1 above)
Comment (elyerry) [reply docs/recipes/recipe-content-analytics/index.md]: 👍

File: docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md (hunk 2 above)
Comment (elyerry) [reply docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md:3]: 👍

File: docs/feature-comparison/index.md [hunk 18]
```diff
@@ -8,12 +8,13 @@ Should you choose [Snowplow Open Source](/docs/getting-started-on-snowplow-open-
...
+| <h3>Create & consume behavioral data</h3> | [Open Source](/docs/getting-started-on-snowplow-open-source/index.md) | [BDP Cloud :new:](/docs/getting-started-with-snowplow-bdp/cloud/index.md) | [BDP Enterprise](/docs/getting-started-with-snowplow-bdp/enterprise/index.md) |
 |:--|:-:|:-:|:-:|
 | [35+ trackers & webhooks](/docs/collecting-data/index.md) | ✅ | ✅ | ✅ |
 | 1st party tracking | ✅ | _coming soon_ | ✅ |
 | [Anonymous data collection](/docs/recipes/recipe-anonymous-tracking/index.md) | ✅ | ✅ | ✅ |
-| [Custom events & entities](/docs/understanding-tracking-design/out-of-the-box-vs-custom-events-and-entities/index.md) | ✅ | _coming soon_ | ✅ |
+| [Custom events & entities](/docs/understanding-tracking-design/out-of-the-box-vs-custom-events-and-entities/index.md) | ✅ | ✅ | ✅ |
+| [Data Structures Builder](/docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md) | ❌ | ✅ | ❌ |
```
Comment (stanch) [inline docs/feature-comparison/index.md:17]:
```suggestion
| [Custom events & entities](/docs/understanding-tracking-design/out-of-the-box-vs-custom-events-and-entities/index.md) | ✅<br/>_(no UI)_ | ✅<br/>_([simplified UI](/docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md))_ | ✅<br/>_(advanced UI)_ |
```

**Review by stanch — CHANGES_REQUESTED** [review]

File: docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md [hunk 19]
```diff
@@ -4,7 +4,7 @@ sidebar_position: 85
 ---
 
 :::info
-The Data Structures Builder is only available for BDP Cloud Customers
+The Data Structures Builder is only available for BDP Cloud customers, for BDP Enterprise users, see [Managing data structures in the console](/docs/understanding-tracking-design/managing-data-structures/index.md), for Open Source users, see [Managing data structures using Iglu](/docs/pipeline-components-and-applications/iglu/igluctl-2/index.md).
```
Comment (stanch) [inline docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md:7]:
```suggestion
The Data Structures Builder is only available for BDP Cloud customers.

For BDP Enterprise customers, see [Managing data structures in the console](/docs/understanding-tracking-design/managing-data-structures/index.md). For Open Source users, see [Managing data structures using Iglu](/docs/pipeline-components-and-applications/iglu/igluctl-2/index.md).
```

File: docs/understanding-tracking-design/managing-data-structures/index.md [hunk 20]
```diff
@@ -4,6 +4,10 @@ date: "2020-02-15"
 sidebar_position: 80
 ---
 
+:::info
+Managing data structures in the console is only available for BDP Enterprise customers, for BDP Cloud users, see [Managing data structures with the Data Structures Builder](/docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md), for Open Source users, see [Managing data structures using Iglu](/docs/pipeline-components-and-applications/iglu/igluctl-2/index.md).
```
Comment (stanch) [inline docs/understanding-tracking-design/managing-data-structures/index.md:8]:
```suggestion
Managing data structures in the console is only available for BDP Enterprise customers.

For BDP Cloud users, see [Managing data structures with the Data Structures Builder](/docs/understanding-tracking-design/managing-data-structures-with-data-structures-builder/index.md). For Open Source users, see [Managing data structures using Iglu](/docs/pipeline-components-and-applications/iglu/igluctl-2/index.md).
```

**Review by stanch — CHANGES_REQUESTED** [review]

**Review by stanch — APPROVED** [review]
